packages/react-composites/src/composites/CallComposite/components/CallReadinessModal.tsx

```diff
@@ -249,15 +245,20 @@ export const CallReadinessModalFallBack = (props: {
     // only way to dismiss this drawer is clicking on allow access which will leads to device permission prompt
   };
 
-  // When permissions are not set, value is undefined, do nothing here
-  // When permissions are set to denied, value is false, show helper screen
-  const showModal = cameraPermissionGranted === false || microphonePermissionGranted === false;
+  const videoState = permissionsState.camera;
+  const audioState = permissionsState.microphone;
+
+  // When permissions are not set, do nothing here
+  // When permissions are set to denied, show helper screen
+  const showModal = videoState === 'denied' || audioState === 'denied';
```

Comment:
Can correct me here but I think we will want to show the modal if videoState is `denied` or `prompt` and not just denied